auth/proxy/firebaseJwt.ts

```diff
@@ -77,14 +79,57 @@ const isJwtPayload = (payload: string | JwtPayload | undefined): payload is JwtP
 
 interface ValidateFirebase {
     token: string
-    firebaseAppIds: string[]
+    configValues: AppConfig
 }
 
-export const validateFirebaseJWT = async ({
-    token,
-    firebaseAppIds,
-}: ValidateFirebase): Promise<void> => {
-    const decodedTokenHeader = decode(token, { complete: true })?.header;
+/**
+ * 
+ * @param payload incoming JWT payload
+ * @param firebaseProjectNumber string representing the Firebase project number
+ * @returns boolean indicating whether the payload has a valid 'iss' claim
+ */
+export const hasValidIss = (
+    payload: object,
+    firebaseProjectNumber: string,
+): payload is JwtPayload =>
+    'iss' in payload &&
+    typeof payload.iss === 'string' &&
+    payload.iss === `https://firebaseappcheck.googleapis.com/${firebaseProjectNumber}`;
+
+    const hasCorrectAudiences = (
+        payload: Record<string, unknown> | JwtPayload,
+        configValues: AppConfig,
+    ): boolean => {
+        const audiences:string[] = [
+            `projects/${configValues.audience}`, 
+            `projects/${configValues.projectId}`];
+        
+        const incomingAudience = payload.aud;
+        
+        return Array.isArray(incomingAudience) 
+            ? incomingAudience.some((aud: string) => audiences.includes(aud)) 
+            : false;
+        
+    }
+
+/**
+ * 
+ * @param payload incoming JWT payload
+ * @param configValues string representing the Firebase project number
+ * @returns boolean indicating whether the payload has a valid 'aud' claim
+ */
+export const hasValidAud = (
```

Comment:
```suggestion
const hasValidAud = (
```

This doesn't need to be exported - it's a function used internally inside the module.